src/Compilers/CSharp/Portable/Symbols/NullableAnnotationExtensions.cs

```diff
@@ -16,49 +16,71 @@ internal static class NullableAnnotationExtensions
 
         /// <summary>
-        /// Check that two nullable annotations are "compatible", which means they could be the same. Return the
-        /// nullable annotation to be used as a result.  This uses the invariant merging rules.
+        /// Return the nullable annotation to use when two annotations are expected to be "compatible", which means
+        /// they could be the same. These are the "invariant" merging rules.
         /// </summary>
-        public static NullableAnnotation EnsureCompatible(this NullableAnnotation a, NullableAnnotation b)
+        public static NullableAnnotation EnsureCompatible(this NullableAnnotation a, NullableAnnotation b) => (a, b) switch
         {
-            if (a.IsOblivious())
-                return b;
-            if (b.IsOblivious())
-                return a;
-            return (a < b) ? a : b;
-        }
+            (NullableAnnotation.Oblivious, _) => b,
+            (_, NullableAnnotation.Oblivious) => a,
+            _ => a < b ? a : b,
+        };
 
         /// <summary>
         /// Merges nullability.
         /// </summary>
-        public static NullableAnnotation MergeNullableAnnotation(this NullableAnnotation a, NullableAnnotation b, VarianceKind variance)
+        public static NullableAnnotation MergeNullableAnnotation(this NullableAnnotation a, NullableAnnotation b, VarianceKind variance) => variance switch
+        {
+            VarianceKind.In => a.Meet(b),
+            VarianceKind.Out => a.Join(b),
+            VarianceKind.None => a.EnsureCompatible(b),
+            _ => throw ExceptionUtilities.UnexpectedValue(variance)
+        };
+
+        /// <summary>
+        /// The attribute (metadata) representation of <see cref="NullableAnnotation.NotAnnotated"/>.
+        /// </summary>
+        public const byte NotAnnotatedAttributeValue = 1;
+
+        /// <summary>
+        /// The attribute (metadata) representation of <see cref="NullableAnnotation.Annotated"/>.
+        /// </summary>
+        public const byte AnnotatedAttributeValue = 2;
+
+        /// <summary>
+        /// The attribute (metadata) representation of <see cref="NullableAnnotation.Oblivious"/>.
+        /// </summary>
+        public const byte ObliviousAttributeValue = 0;
+
+        /// <summary>
+        /// The mapping from NullableAnnotation to the attribute (metadata) representations.
+        /// </summary>
+        public static byte ToAttributeValue(this NullableAnnotation self) => self switch
+        {
+            NullableAnnotation.NotAnnotated => NotAnnotatedAttributeValue,
+            NullableAnnotation.Annotated => AnnotatedAttributeValue,
+            NullableAnnotation.Oblivious => ObliviousAttributeValue,
+            _ => throw ExceptionUtilities.UnexpectedValue(self)
+        };
+
+        /// <summary>
+        /// The mapping from the attribute (metadata) representations to NullableAnnotation.  This
+        /// method is not actually used, but is here to document the mapping.
```

Comment:
>method is not actually used, but is here to document the mapping. [](start = 12, length = 65)

I do not think there is a value in keeping the dead code in the compiler. The mapping can be documented in a comment if we need special documentation at all.

Reply:
I agree with @AlekseyTs, we don't need to keep commented out code.

Besides, the explicit `const` values above already indicate a potential difference between the attribute values and the `enum` values, and the mapping between the two is simple.

---
In reply to: [274676177](https://github.com/dotnet/roslyn/pull/34909#discussion_r274676177) [](ancestors = 274676177)